src/pytest_html/nextgen.py

```diff
@@ -291,14 +313,11 @@ def pytest_runtest_logreport(self, report):
             test_id += f"::{report.when}"
         data["testId"] = test_id
 
-        # Order here matters!
-        log = report.longreprtext or report.capstdout or "No log output captured."
-        data["log"] = _handle_ansi(log)
         data["result"] = _process_outcome(report)
         data["extras"] = self._process_extras(report, test_id)
 
-        self._report.add_test(data)
-        self._generate_report()
+        if self._report.add_test(data, report):
+            self._generate_report()
```

Comment:
Only generate the report if we actually made a change (added a test in this case).